### ISSUE TITLE
bounded_graph argument of the get_area() method

### ISSUE THREAD

**Issue by brbisheng** [ManimCommunity/manim#3470]
perhaps I am wrong, but it seems to me that the **bounded_graph** argument of the **get_area()** method should have its behavior updated, since class **ParametricFunction** has no longer the **t_min** and **t_max** attributes.

Errors as below:

File /usr/local/lib/python3.8/site-packages/manim/mobject/graphing/coordinate_systems.py:1331, in CoordinateSystem.get_area(self, graph, x_range, color, opacity, bounded_graph, **kwargs)
   1329     a, b = x_range
   1330 if bounded_graph is not None:
-> 1331     if bounded_graph.t_min > b:
   1332         raise ValueError(
   1333             f"Ranges not matching: {bounded_graph.t_min} < {b}",
   1334         )
   1335     if bounded_graph.t_max < a:

AttributeError: 'function' object has no attribute 't_min'

**Thank you for the brilliant Manim.**

**Comment by behackl** [comment]
Indeed, thanks for the report! Fixing this would be a good first contribution, in case you are interested. :-)

**Comment by brbisheng** [comment]
Thank you also for the timely reply, I could try.. 😏 (although I am quite unfamiliar with the process.)

**Comment by behackl** [comment]
Feel free to try! We have a written guide to get started at https://docs.manim.community/en/stable/contributing/development.html -- and we've had a short session on contribution at our latest community meeting, you can find that to follow along here: https://youtu.be/JbUjqf4GVdM.

**Comment by brbisheng** [comment]
Thank you. 😎

**Comment by vitesse206** [comment]
I would like to take a shot at it aswell. Is this possible (I am new to open-source)?

**Comment by brbisheng** [comment]
Hello, everyone, it is ok, please work on it if you would like to!

**Comment by MontroyJosh** [comment]
In the functions.py file from manim/mobject/graphing/function.py, there appears to be an initialization of t_min on line 120. Does this issue persist? @brbisheng 
<img width="458" alt="Screenshot 2024-04-01 at 1 29 35 PM" src="https://github.com/ManimCommunity/manim/assets/122334909/fdb4a83f-55fd-4a3a-bb95-914a58c9f7b5">

**Comment by acremers22** [comment]
I'm looking into a solution for this issue.

**Comment by acremers22** [comment]
@behackl @brbisheng Following up.. we are still looking into this issue but are wondering if you have any additional details? To us, it seems that t_min and t_max are initialized in the ParametricFunction class in functions.py? Does this issue persist?

**Comment by anjaliagr0309** [comment]
To resolve this issue, you will need to update the get_area() method to reflect the current implementation of the ParametricFunction class.
try this- 
`class CoordinateSystem:
    def get_area(self, graph, x_range, color=None, opacity=1.0, bounded_graph=None, **kwargs):
        a, b = x_range
        if bounded_graph is not None:
            # Update this part to check the appropriate attributes or methods
            # For example, if bounded_graph now uses `start` and `end` instead of `t_min` and `t_max`
            if hasattr(bounded_graph, 'start') and hasattr(bounded_graph, 'end'):
                if bounded_graph.start > b:
                    raise ValueError(f"Ranges not matching: {bounded_graph.start} > {b}")
                if bounded_graph.end < a:
                    raise ValueError(f"Ranges not matching: {bounded_graph.end} < {a}")
            else:
                raise AttributeError("bounded_graph does not have the expected attributes `start` and `end`.")

        # Rest of the method implementation...
`

**Comment by antil-samarth** [comment]
Hi
Has this been resolved?

**Comment by chopan050** [comment]
As @MontroyJosh mentioned, `ParametricFunction` does, indeed, have `t_min` and `t_max` attributes. I have used the `.get_area()` method with `bounded_graph` and it worked correctly.

The actual error happens here:
```
AttributeError: 'function' object has no attribute 't_min'
```

According to this error, it seems **you attempted to pass a literal function as `bounded_graph`: the ones you define with `def` or `lambda`, like, for example, `lambda x: np.exp(-x)`**. The expected object was, instead, a `ParametricFunction` Mobject (or `None` - there should be a PR fixing the type hint of `bounded_graph`), like, for example, the result of `axes.plot(lambda x: np.exp(-x), x_range=[0, 2])`.

For that reason, I'm closing this issue. Thanks for looking into this!
